### PR TITLE
Improve compilation in AppVeyor  using clcache

### PR DESCRIPTION
**Pull request by tadeu**

### CODE REVIEW

**Comment by tadeu** [comment]
I just rebased this over #41, so that one should be merged first, then this one will only contain its own changes

File: environment.devenv.yml [hunk 1]
```diff
@@ -29,6 +29,8 @@ dependencies:
 {% endif %}
 {% if sys.platform.startswith('linux') or sys.platform.startswith('osx') %}
   - ccache
+{% else %}
```
Comment (nicoddemus) [inline environment.devenv.yml]: You can remove this if/else now that `clcache` is used in all platforms. 😁

**Comment by nicoddemus** [comment]
I triggered an AppVeyor rebuild to see how much `clcache` gains us.

**Comment by tadeu** [comment]
😅 
Well, it's within the 1 Gb limit from the free tier. I'm thinking on caching the conda env next ;)

**Comment by tadeu** [comment]
Down to 10 minutes each Windows job ;)

**Comment by allanleal** [comment]
This is great @tadeu - thanks a lot!

**Pull request merged**